miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -57,6 +64,31 @@ abstract class MiniAppMessageBridge {
         }
     }
 
+    private fun onRequestCustomPermissions(callbackObj: CallbackObj) {
+        try {
+            val customPermissionParam = Gson().fromJson<CustomPermission>(
+                callbackObj.param.toString(),
+                object : TypeToken<CustomPermission>() {}.type
+            )
+
+            val permissions: ArrayList<MiniAppCustomPermissionType> = arrayListOf()
+            customPermissionParam.permissions.forEach { permission ->
+                MiniAppCustomPermissionType.getValue(permission)?.let { permissions.add(it) }
+            }
+
+            requestCustomPermissions(
+                permissions.toList()
+            ) { grantResult ->
+                onRequestCustomPermissionsResult(
+                    callbackId = callbackObj.id,
+                    grantResult = grantResult
```

Comment:
We need to align on iOS/Android for the format that's returned to the Mini App. Probably something like this:
```
{
  permissions: [
    "rakuten.miniapp.USER_NAME": "ALLOWED",
    "rakuten.miniapp.PROFILE_PHOTO": "DENIED"
  ]
}
```

@ts-leojoseph-r Have you worked on the return format for iOS yet? What were you thinking for this?

Reply:
@corycaywood That current idea I was planning to return is 
```
{
  permissions: [
    {"name":"rakuten.miniapp.USER_NAME", "isGranted": "ALLOWED"},
    {"name":"rakuten.miniapp.PROFILE_PHOTO", "isGranted": "DENIED"},
    {"name":"rakuten.miniapp.PHONE_NUMER", "isGranted": "PERMISSION_NOT_AVAILABLE"}
  ]
}
```

And the request will be like,

```
[
    {"name":"rakuten.miniapp.USER_NAME", "description": "Mini app needed User name for registration"},
    {"name":"rakuten.miniapp.PROFILE_PHOTO", "description": "Mini app needed Profile pic for registration"},
    {"name":"rakuten.miniapp.PHONE_NUMER", "description": "Mini app needed Phone number for offers"}
]
```

Reply:
@ts-leojoseph-r Okay, I think that looks good.

Reply:
Do we need to prepare this response inside SDK? or, inside HostApp? 
I was planning to do it in HostApp.

Reply:
@khairul-alam-licon This response should be prepared inside the SDK. The response must match what we define in the JavaScript SDK, so we can't depend on the Host App to correctly define it.